### PR TITLE
Added Vim command for `create_element_from_path.py`

### PR DESCRIPTION
**Pull request by AVDiv**
#48 
Added Vim/NeoVim command for `create_element_from_path.py` with instructions to setup.
Please notify if any changes are needed to be done.

### CODE REVIEW

**Comment by AVDiv** [comment]
Hey, I just signed the CLA.

**Comment by qpavsmi** [comment]
Thank you for your contribution! Will try out the instructions now and come back to you on approval or improvement ideas!

**Comment by qpavsmi** [comment]
@cla-bot

**Comment by qpavsmi** [comment]
> Thank you for your pull request and welcome to our community.
> 
> We require contributors to sign our Contributor License Agreement (CLA), and it seems that the following contributors have not yet signed it: @AVDiv.
> 
> **In order for us to review and merge your code, please [sign the CLA at meetiqm.com](https://meetiqm.com/developers/clas/)**.
> 
> Once you have signed the CLA, write a comment here and we'll verify. Please note that it may take some time for us to verify it.

Check

**Comment by qpavsmi** [comment]
@cla-bot check

**Comment by AVDiv** [comment]
Any improvements to be done? @qpavsmi

File: util/create_element_from_path.py [hunk 1]
```diff
@@ -81,7 +81,26 @@
...
+#   2. Add the following lines to your configuration file:
+#       " Define a kqc command
+#       command! -nargs=0 kqc execute "!klayout -e -rm " . expand("%:p:h") . "/util/create_element_from_path.py -rd element_path=" . expand("%:p") | redraw!
+#       " Create a mapping to run the kqc command
+#       nnoremap <leader>kqc :kqc<CR>
+#
+#   3. Save the configuration file and restart Vim or NeoVim for the changes to take effect.
+#       Now, you can use the mapping <leader>kqc (where <leader> is a customizable key, often \ by default) while editing a Python file containing the element to create. 
+#       Pressing the mapping will execute the :kqc command, which runs the external tool command to open the file in KLayout.
+#
```
[the remark this answers is not in the document]
Comment (qpavsmi) [reply util/create_element_from_path.py:103]: Above are proposed changes

**Review by qpavsmi — CHANGES_REQUESTED** [review]

**Comment by qpavsmi** [comment]
> Any improvements to be done? @qpavsmi

Hi, sorry, I thought my review went public but apparently it didnt: https://github.com/iqm-finland/KQCircuits/pull/55#discussion_r1224041294

File: util/create_element_from_path.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AVDiv) [reply util/create_element_from_path.py:103]: Nice I'll check on that ASAP.

**Comment by AVDiv** [comment]
> > Any improvements to be done? @qpavsmi
> 
> Hi, sorry, I thought my review went public but apparently it didnt: [#55 (comment)](https://github.com/iqm-finland/KQCircuits/pull/55#discussion_r1224041294)

No issues👍

**Comment by AVDiv** [comment]
Right, done with it. Anything else to improve? @qpavsmi

**Comment by qpavsmi** [comment]
> Right, done with it. Anything else to improve? @qpavsmi

Thank you for the update, will check it when I get home

**Comment by qpavsmi** [comment]
> > Right, done with it. Anything else to improve? @qpavsmi
> 
> Thank you for the update, will check it when I get home

Looking at the diffs on my phone they look good, thank you! Please squash the three commits into one commit with a descriptive commit message (message for first commit was good enough).

**Comment by AVDiv** [comment]
> > > Right, done with it. Anything else to improve? @qpavsmi
> > 
> > 
> > Thank you for the update, will check it when I get home
> 
> Looking at the diffs on my phone they look good, thank you! Please squash the three commits into one commit with a descriptive commit message (message for first commit was good enough). It will probably require force push.

Okay

**Comment by AVDiv** [comment]
I think starting off with a new PR would be nice, right?

**Comment by qpavsmi** [comment]
> I think starting off with a new PR would be nice, right?

Actually dont worry about

**Comment by AVDiv** [comment]
> > I think starting off with a new PR would be nice, right?
> 
> Actually dont worry about

OK

**Comment by qpavsmi** [comment]
> > > I think starting off with a new PR would be nice, right?
> > 
> > 
> > Actually dont worry about
> 
> OK

I updated the last message :) anyway you could check some static analysis failures (just pylint stuff) here: https://github.com/iqm-finland/KQCircuits/actions/runs/5231064161/jobs/9444987678?pr=55 some lines were too long, so those can be fixed before merge

**Comment by AVDiv** [comment]
> > > > I think starting off with a new PR would be nice, right?
> > > 
> > > 
> > > Actually dont worry about
> > 
> > 
> > OK
> 
> I updated the last message :) anyway you could check some static analysis failures (just pylint stuff) here: https://github.com/iqm-finland/KQCircuits/actions/runs/5231064161/jobs/9444987678?pr=55 some lines were too long, so those can be fixed before merge

yep, just noticed it & now I'm fixing it.

**Comment by AVDiv** [comment]
Right fixed those

**Review by qpavsmi — COMMENTED** [review]
Approving by comment, and will mark you as a receiver for the bounty. Depending on whether I'll have time tomorrow or early next week I will merge this as part of our official branch, there is a process for it and I need to be careful. Thank you very much for the contribution and good job!

**Comment by AVDiv** [comment]
> Approving by comment, and will mark you as a receiver for the bounty. Depending on whether I'll have time tomorrow or early next week I will merge this as part of our official branch, there is a process for it and I need to be careful. Thank you very much for the contribution and good job!

Thank you, grateful to work with you.🙌

**Comment by AVDiv** [comment]
hi @qpavsmi ,
Can you also mark the issue as done on [unitaryhack](https://unitaryhack.dev)

**Comment by qpavsmi** [comment]
> hi @qpavsmi , Can you also mark the issue as done on [unitaryhack](https://unitaryhack.dev)

Theyre supposed to have an automatic system for that, not sure why it hasnt noticed your bounty yet. Ive reached out the unitaryhack team about this.

**Comment by qpavsmi** [comment]
> > hi @qpavsmi , Can you also mark the issue as done on [unitaryhack](https://unitaryhack.dev)
> 
> Theyre supposed to have an automatic system for that, not sure why it hasnt noticed your bounty yet. Ive reached out the unitaryhack team about this.

Now its marked as done in UH site

**Comment by AVDiv** [comment]
> Theyre supposed to have an automatic system for that, not sure why it hasnt noticed your bounty yet. Ive reached out the unitaryhack team about this.

Yep, I too thought it works that way looks like there is an issue, anyways thank you for marking this manually.😁

**Review by qpavsmi — APPROVED** [review]

**Comment by qpavsmi** [comment]
Merged manually, closing PR

**Pull request closed without merging**